lib/ruby_lsp/requests/completion_resolve.rb

```diff
@@ -0,0 +1,37 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    class CompletionResolve < Request
+      extend T::Sig
+      include Requests::Support::Common
+
+      MAX_DOCUMENTATION_ENTRIES = 10
+
+      sig { params(index: RubyIndexer::Index, item: T::Hash[Symbol, T.untyped]).void }
+      def initialize(index, item)
+        super()
+        @index = index
+        @item = item
+      end
+
+      sig { override.returns(Interface::CompletionItem) }
+      def perform
+        entries = @index.find(@item[:label])
+        Interface::CompletionItem.new(
+          **T.unsafe({
+            **@item.transform_keys { |k| k.to_s.gsub(/(.)([A-Z])/, '\1_\2').downcase.to_sym },
```

Comment:
It wasn't clear to me from the [LSP spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_completion) if all properties need to be included in the response, or if we can send only the `label_details` and `documentation`.

The spec says:

> All other properties (usually sortText, filterText, insertText and textEdit) must be provided in the [textDocument/completion](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_completion) response and must not be changed during resolve.

The code works as intended either way, but including all passed properties _in addition_ to `label_details` and `documentation` required this ugly camel case to snake case conversion.

Reply:
I don't think we need to include all of the previous properties. Label details and documentation are probably enough.